subscription/notify-test-lambda/src/notify-test-lambda.py

```diff
@@ -0,0 +1,86 @@
+import json
+import boto3
+from botocore.exceptions import ClientError
+
+# This lambda is to be used in SIT and WL to test the ingest subscription notification process.
+# This is a test lambda that gets triggered off of the interal CMR load balancer.
+# It receives a notification generated by the ingest notification lambda. 
+# The notification is stored in a file in an S3 bucket. Ussers can tunnel into the CMR internal only 
+# load balancer and run curl http://localhost:8081/notification/tester.  Make sure to use the correct port number
+# that you used in your tunnel.
+
+s3 = boto3.client('s3')
+BUCKET_NAME = 'cmr-notification-http-tester-bucket'
+FILE_NAME = 'notification-message.json'
+
+
+# handler(event: Dict[str, Any], context: Any) -> None:
+def handler(event, context):
+    """This handler is invoked from the CMR internal load balancer from an http request.
+       If method is get, then read the contents of the file in the bucket and return the file contents.
+       The caller can then verify that the notfication that was sent by the ingest notification lambda is correct.
+       If method is post, then save the contents of the event body in the file located in the bucket.
+       The contents can be recalled by the get method to verify the sent notification."""
+
+    method = event['httpMethod']
+
+    if method == 'GET':
+        return handle_get()
+    elif method == 'POST':
+        body = event['body']
+        return handle_post(body)
+    else:
+        return {
+            'statusCode': 405,
+            'body': json.dumps('Method Not Allowed')
+        }
+
+def handle_post(body):
+    """This method save the contents of the request body into a file."""
+    try:
+        # Parse the incoming JSON data
+        body = json.loads(body)
+        
+        # Save the data to S3
+        s3.put_object(
+            Bucket=BUCKET_NAME,
+            Key=FILE_NAME,
+            Body=json.dumps(body),
+            ContentType='application/json'
+        )
+        
+        return {
+            'statusCode': 200,
+            'body': json.dumps('Data saved successfully')
+        }
+    except Exception as e:
+        print(f"Error in POST: {str(e)}")
+        return {
+            'statusCode': 500,
```

Comment:
is this a general catch all exception and thus 500 is good, or is there something more specific going on and another error is better?

Reply:
It just a testing tool that saves to a file in an S3 bucket. I could figure out what else could be thrown, but I don't think it is worth it.